examples/lib/stories/system/step_engine_example.dart

```diff
@@ -0,0 +1,138 @@
+import 'dart:async';
+import 'dart:math';
+
+import 'package:flame/collisions.dart';
+import 'package:flame/components.dart';
+import 'package:flame/effects.dart';
+import 'package:flame/events.dart';
+import 'package:flame/game.dart';
+import 'package:flame/palette.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+class StepEngineExample extends FlameGame
+    with HasCollisionDetection, HasKeyboardHandlerComponents {
+  static const description = '''
+    This example demonstrates how the game can be advanced frame by frame using
+    stepEngine method.
+
+    To pause and un-pause the game anytime press the `P` key. Once paused, use
+    the `S` key to step by one frame.
+
+    Up arrow and down arrow can be used to increase or decrease the step time.
+  ''';
+
+  // Fixed resolution of the game.
+  static final Vector2 _visibleSize = Vector2(320, 180);
+  double _stepTimeMultiplier = 1;
+  static const _stepTime = 1 / 60;
+
+  @override
+  Color backgroundColor() => BasicPalette.darkGreen.color;
+
+  @override
+  Future<void> onLoad() async {
+    final carSprite = await Sprite.load('Car.png');
+    final car = SpriteComponent(
+      sprite: carSprite,
+      anchor: Anchor.center,
+      angle: -pi / 10,
+      position: Vector2(0, _visibleSize.y / 3),
+      children: [CircleHitbox()],
+    );
+
+    final componentsToAdd = _createCircularDetectors();
+    componentsToAdd.add(PositionComponent(children: [car, _rotateEffect]));
+
+    final world = World(children: componentsToAdd);
```

Comment:
```suggestion
    final world = World(
      children: [
        ..._createCircularDetectors(),
        PositionComponent(children: [car, _rotateEffect]),
      ]
    );
```
Maybe slightly easier to read?